core/trino-main/src/main/java/io/trino/sql/planner/PlanOptimizers.java

```diff
@@ -864,8 +865,9 @@ public PlanOptimizers(
                         statsCalculator,
                         costCalculator,
                         ImmutableSet.<Rule<?>>builder()
-                                // Run RemoveEmptyDeleteRuleSet and RemoveEmptyTableExecute after table scan is removed by PickTableLayout/AddExchanges
+                                // Run RemoveEmptyDeleteRuleSet, RemoveEmptyUpdate and RemoveEmptyTableExecute after table scan is removed by PickTableLayout/AddExchanges
                                 .addAll(RemoveEmptyDeleteRuleSet.rules())
+                                .add(new RemoveEmptyUpdate())
```

Comment:
for Delete we have a rule set (since we match empty delete with and without an exchange).
wonder whether this is some left over (or otherwise: why don't we need same for Update?)

Reply:
followed up with https://github.com/trinodb/trino/pull/13794